### PR TITLE
Install Kani into the action, instead of using a docker

### PR DESCRIPTION
**Pull request by danielsn**
### Description of changes: 

Currently, the action requires a docker, which is not yet published. Instead, as a temporary solution, just install the dependencies every time. 

### Resolved issues:

Resolves #1589 

### Related RFC:

Optional #ISSUE-NUMBER.

### Call-outs:

- Installing the dependencies takes about 1-1.5 minutes according to some initial testing.

### Testing:

* How is this change tested? Made a repo with Kani proofs, ran this action on it.

* Is this a refactor change? No

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -13,12 +13,20 @@ inputs:
   command:
     description: 'Command to run.'
     required: false
-    default: 'cargo kani --workspace'
+    default: 'cargo-kani'
```
Comment (tedinski) [inline action.yml:16]: Is this default change desireable?

**Review by celinval — COMMENTED** [review]
I do have one concern regarding mixing the repositories. I was puzzled about having the `action.yml` in the root of the repository. So I decided to read a bit more about actions. This is basically the first thing in their documentation:

> When you plan to publish your action to GitHub Marketplace, you'll need ensure that the repository only includes the metadata file, code, and files necessary for the action.

**Comment by tedinski** [comment]
That's a pretty good point. We should have a `kani-action` repo or something?

File: action.yml [hunk 2]
```diff
@@ -13,12 +13,20 @@ inputs:
...
-  args:
-    - bash
-    - '-c'
-    - 'set -e; export HOME=/root USER=$(id -nu) PATH=/root/.cargo/bin:$PATH; source $HOME/.bashrc; ${{ inputs.command }}'
+  using: "composite"
+  steps:
+    - name: Install Kani
+      shell: bash
+      run: |
+        export KANI_VERSION="0.11.0"; \
```
Comment (celinval) [inline action.yml:24]: Should we retrieve the version from the repository instead? I was also wondering if we should allow users to choose the version they want to use so they can control when to update Kani.

Comment (danielsn) [reply action.yml:24]: My thought was action version matches kani-version. So we have kani-action@0.11 matches kani 0.11

File: action.yml (hunk 1 above)
Comment (danielsn) [reply action.yml]: Fixed

**Comment by celinval** [comment]
@danielsn Can we close this PR since we have published the Action in a different repo?

**Pull request closed without merging**